Implement tensor.size(Dimname), tensor.stride(Dimname)

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23172 Implement tensor.set_names_, tensor.names setter
* #23106 Named inference rule for torch.prod
* #23081 Implement named inference rule for torch.sum
* #23075 Refactor named inference rule for reductions
* **#22989 Implement tensor.size(Dimname), tensor.stride(Dimname)**

Test Plan
- [namedtensor ci]

Differential Revision: [D16364437](https://our.internmc.facebook.com/intern/diff/D16364437)

zou3519 committed:
Implement tensor.size(Dimname), tensor.stride(Dimname)

Test Plan
- [namedtensor ci]

nairbv commented on tools/autograd/templates/python_variable_methods.cpp:
```diff
@@ -88,6 +89,11 @@ static PyObject * THPVariable_size(PyObject* self, PyObject* args, PyObject* kwa
+    if (jit::tracer::isTracing()) {
+      TORCH_INTERNAL_ASSERT("NYI: Named tensors w/ JIT");
+    }
+    return wrap(self_.size(r.dimname(0)));
```
needs an ifdef or?

zou3519 replied:
Uh oh... thanks for the catch

zou3519 committed:
Update on "Implement tensor.size(Dimname), tensor.stride(Dimname)"

Implement tensor.size(Dimname), tensor.stride(Dimname)

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 22989 gh/zou3519/69/head

zou3519 committed:
Update on "Implement tensor.size(Dimname), tensor.stride(Dimname)"

Implement tensor.size(Dimname), tensor.stride(Dimname)

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 22989 gh/zou3519/69/head

zou3519 commented:
@nairbv this should be good for another pass through
